### PR TITLE
[Add] Adds the accentless trait

### PR DESCRIPTION
**Pull request by OctoRocket**
## About the PR

I'm making this PR to try to get the accentless trait in the game yet again. Previously, it had a lot of support yet I gave in too quickly to suggestions I couldn't make.

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added an accentless trait.

### CODE REVIEW

**Comment by OctoRocket** [comment]
For some reason, it crashes when trying to remove the dwarf accent (Replacement accent component with accent field dwarf) via RemComp because of an out of bounds error. I'm not sure what is causing this, and it seems to be an issue with the RemComp function itself.

**Comment by OctoRocket** [comment]
Thanks to Mirrorcult's insights, it should now be working!

File: Content.Shared/Traits/Assorted/AccentlessComponent.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Traits.Assorted;
+
+/// <summary>
+/// This is used for the accentless trait
+/// </summary>
+[RegisterComponent]
+public sealed class AccentlessComponent : Component
+{
```
Comment (metalgearsloth) [inline Content.Shared/Traits/Assorted/AccentlessComponent.cs:10]: Needs to be NetworkedComponent

File: Content.Shared/Traits/Assorted/AccentlessComponent.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    /// <summary>
+    ///     The accents removed by the accentless trait.
+    /// </summary>
+    [DataField("removes"), ViewVariables(VVAccess.ReadWrite)]
+    public ComponentRegistry RemovedAccents = new();
```
Comment (metalgearsloth) [inline Content.Shared/Traits/Assorted/AccentlessComponent.cs:15]: Mark it as required.

**Comment by OctoRocket** [comment]
That's an unfortunate time for the review to come in...

**Comment by Emisse** [comment]
its fine just do it when you can

**Pull request merged**